plugins/source/gcp/resources/services/compute/zones.go

```diff
@@ -46,13 +50,12 @@ func fetchZones(ctx context.Context, meta schema.ClientMeta, parent *schema.Reso
 	it := gcpClient.List(ctx, req, c.CallOptions...)
 	for {
 		resp, err := it.Next()
-		if err == iterator.Done {
-			break
-		}
 		if err != nil {
+			if errors.Is(err, iterator.Done) {
```

Comment:
Can we not do this refactoring in this PR? I checked a few resources and they use the current pattern of not nesting the check for `iterator.Done`.